src/components/Modals/MobileWelcome/views/Notice.tsx

```diff
@@ -17,6 +19,11 @@ export const Notice = () => {
   const handleRecoveryClick = () => {
     handleClose()
     open({ preventClose: true })
+    store.dispatch(preferences.actions.setWelcomeModal({ show: false }))
+  }
+  const handleDismissClick = () => {
```

Comment:
`useCallback()`, I'm not so concerned if existing event handlers aren't tackled in PRs, but any event handlers or expressions passed down as props which are not memoized make it harder for us to get full memoization across the app (https://github.com/shapeshift/web/pull/2699) in